packages/markers/src/browser/marker-tree.ts

```diff
@@ -131,6 +132,7 @@ export namespace MarkerNode {
 export interface MarkerInfoNode extends UriSelection, SelectableTreeNode, ExpandableTreeNode {
     parent: MarkerRootNode;
     numberOfMarkers: number;
+    severity?: DiagnosticSeverity;
```

Comment:
I'm not sure I follow this change, I don't believe `MarkerInfoNode` which represents a resource (ex: `package.json`) should have a severity. Logically I don't think `severity` applies to a resource, only a diagnostic itself.

Reply:
The MarkerInfoNodes/resources will be sorted based on the most severe diagnostic marker they contain. The severity attribute in the MarkerInfoNode stores the severity of its most severe diagnostic marker.
This way, when comparing resources, I can use this attribute instead of iterating through all diagnostic markers of the resource.

As improvement I could:
1. try to find a more fitting/self-explaining name and/or data type
2. remove the severity here and iterate through the diagnostic markers for each comparison of resources.
3. try to find a more suited place to store this kind of information.

Do you have any preferences or other ideas?

Reply:
@LukasBoll I don't believe resources should be sorted based on severity, they should be sorted based on their `uri` (https://github.com/eclipse-theia/theia/issues/7318). Sorting based on the highest severity marker of a resource will yield unexpected results compared to vscode.

Reply:
I think there is an inconsistency between #7318 and #8983
I thought you agreed on implementing it based on severity. (See the last two comments of #8983)

Reply:
In the current implementation you actually immediately see all changes in the first level.

Reply:
I see, there are two things happening:
- sort resources based on highest severity
- sort resources by `uri` for a given severity (ex: both have `error`).

Both should be supported by the fix to have the proper behavior.

Reply:
Yes, this is how it is implemented.